#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_gtk-immodule-install-dir.dpatch by Ming Hua <minghua@rice.edu>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Change the gtk immodule installation directory from
## DP: /usr/lib/gtk-2.0/immodules/ to
## DP: /usr/lib/gtk-2.0/$(GTK_BINARY_VERSION)/immodules/.
## DP:
## DP: The GTK+ IM module should be installed in the ABI-specific
## DP: directory instead of the general one to avoid ABI mismatch during
## DP: GTK+ ABI upgrade.
## DP:
## DP: Since GTK+ 2.10, Debian has changed the way to handle the ABI for
## DP: IM modules and GdkPixbuf loaders.  For IM modules, the
## DP: /usr/sbin/update-gtk-immodules tool is deprecated and the IM
## DP: module information is not stored in /etc/gtk-2.0/gtk.immodules
## DP: anymore.  The debhelper program dh_gtkmodules should be used
## DP: instead to update the IM module information.
## DP:
## DP: NOTE:  This patch only changes extras/gtk_immodule/Makefile.am,
## DP: and the change necessary for extras/gtk_immodule/Makefile.in is
## DP: incorporated in 90_relibtoolize.dpatch.  If the relibtoolize
## DP: patch is removed, this patch should change Makefile.in as well.

@DPATCH@
diff -urNad scim-1.4.6~/extras/gtk2_immodule/Makefile.am scim-1.4.6/extras/gtk2_immodule/Makefile.am
--- scim-1.4.6~/extras/gtk2_immodule/Makefile.am	2007-04-10 03:07:55.000000000 -0500
+++ scim-1.4.6/extras/gtk2_immodule/Makefile.am	2007-04-23 13:42:21.000000000 -0500
@@ -32,7 +32,7 @@
 
 noinst_HEADERS = gtkimcontextscim.h
 
-moduledir = @GTK_LIBDIR@/gtk-2.0/immodules
+moduledir = @GTK_LIBDIR@/gtk-2.0/@GTK_BINARY_VERSION@/immodules
 
 module_LTLIBRARIES = $(CONFIG_GTK2_IMMODULE)
 
